tensorflow/compiler/mlir/tosa/transforms/legalize_common.cc

```diff
@@ -135,7 +135,8 @@ llvm::Optional<Value> convertPackOp(PatternRewriter& rewriter, Operation* op,
 
   // Check for ranked tensor type.
   if (!result_type) {
-    op->emitOpError("PackOp: result type not ranked tensor");
+    (void)rewriter.notifyMatchFailure(op,
+        "PackOp: result type not ranked tensor");
```

Comment:
Nit: drop "PackOp:" , the rest follows the error conventions and the failure will point to op directly so there shouldn't be need to repeat op type.